### PR TITLE
feat(media): show Missing status on unmonitored items

### PR DESCRIPTION
**Pull request by arsfeld**
## Problem

Unmonitored movies never showed a Missing badge and never appeared under the library's Missing filter. Anyone who adds titles as a reminder and downloads them manually saw an empty Missing list, which is exactly the workflow that needs it most.

The cause is that `Mydia.Media.get_media_status/1` treated monitoring and availability as one axis. For an unmonitored movie it returned `:not_monitored` and never looked at the files:

```elixir
def get_media_status(%MediaItem{type: "movie", monitored: false} = media_item) do
  file_count = length(media_item.media_files)
  {:not_monitored, %{has_files: file_count > 0, file_count: file_count}}
end
```

That single classifier drives both the card badge and the Missing filter, so an unmonitored item could never match `:missing`.

## Approach

A new `Mydia.Media.AvailabilityStatus` struct splits the two axes. `state` answers "do we have this?" and `monitored` answers "are we trying to get it?". `:not_monitored` is gone from the state vocabulary.

Unmonitored items now render their real availability with the same icon and hue at reduced weight (`badge-outline opacity-60`), so Missing still reads as missing without lighting up a library of deliberately ignored titles. Monitoring stays visible through the bookmark toggle already on every card and row, and the tooltip gains a "Not monitored" suffix so a faded badge explains itself.

Three branches were collapsing into `:not_monitored`, and all three are fixed:

1. Unmonitored movies
2. Unmonitored shows
3. Monitored shows whose episodes are all unmonitored, which reported "Not Monitored" despite the show being monitored

Episode rows on the show detail page get the same treatment, so a show card and its own episode table cannot disagree.

For series, episodes are classified over the monitored episodes when there are any and over every episode when there are none, so the x/y counts always have a meaningful denominator.

## Filter behaviour

The Missing filter now matches unmonitored items. The previous result set is still reachable without new UI, since the library already has a separate monitoring select: Missing + Monitored gives exactly what Missing gave before.

## Dead code removed

Mapping the call sites turned up code that nothing renders, all of it deleted here rather than ported to the new struct:

- `library_card/1` and `library_row/1` in `library_components.ex`, plus every private helper in that file, which were reachable only from those two functions. `index.html.heex` has its own inline markup.
- `Mydia.LibraryItem` and its three implementations (`media_item.ex`, `album.ex`, `book.ex`). Nothing outside `lib/mydia/library_item*` referenced the behaviour. It existed to feed the two components above.

Incidentally this retires a latent bug: `album.ex` and `book.ex` both returned a `:complete` status that had no matching colour clause and would have raised if ever rendered.

The green "unmonitored but has files" indicator is also gone, since such an item now renders a muted Downloaded badge directly rather than stacking two badges.

## Testing

- `AvailabilityStatus` unit tests over colour, icon and label for every state in both monitoring states
- `get_media_status/1` matrix covering movie and series, monitored and unmonitored, with no files, some, all, upcoming and downloading, plus the zero-monitored-episodes branch and the empty-episodes edge case
- `EpisodeStatus` tests updated, including the two that previously asserted `:not_monitored`
- LiveView coverage that the Missing filter returns an unmonitored fileless movie, that Missing + Monitored still narrows to monitored only, and that an unmonitored show with no episode files matches
- Episode badge rendering tests asserting the muted classes

Full suite: 6804 tests, 0 failures. `mix credo --strict` clean.

### CODE REVIEW

File: lib/mydia/media.ex [hunk 1]
```diff
@@ -999,98 +999,77 @@ defmodule Mydia.Media do
...
+      end
 
-      {status, %{downloaded: downloaded_count, total: total_monitored}}
-    end
+    %AvailabilityStatus{
+      state: state,
+      # A monitored show with every episode unmonitored is not chasing anything, so it
+      # renders muted rather than claiming a pursuit that will never happen.
+      monitored: media_item.monitored and monitored_episodes != [],
+      downloaded: downloaded_count,
```
[the remark this answers is not in the document]
Comment (arsfeld) [reply lib/mydia/media.ex:1072]: Good catch, fixed in 9b931fd2.

The rule now distinguishes "no episodes yet" from "episodes exist but none monitored":

```elixir
monitored: media_item.monitored and (episodes == [] or monitored_episodes != []),
```

A show awaiting metadata has nothing contradicting its own flag, so it stays solid; a show whose episodes all exist and are unmonitored still mutes.

Worth noting the existing empty-episodes test did cover that input but only asserted `state` and the counts, never `monitored`, which is why it slipped through. Added three tests pinning both sides of the boundary.
